File: pub-sub/subscriber.rb
require "bunny"

pid = Process.pid

puts "[#{pid}] Enter server IP (localhost): "
host = gets.chomp
host = "localhost" if host.empty?

conn = Bunny.new(host: host)
conn.start

ch  = conn.create_channel
x   = ch.fanout("logs")
q   = ch.queue("", exclusive: true)

q.bind(x)

puts "[#{pid}] Waiting for logs. To exit press CTRL+C"

begin
  q.subscribe(:block => true) do |delivery_info, properties, body|
    puts "[#{pid}] #{body}"
  end
rescue Interrupt
  ch.close
  conn.close
  puts "[#{pid}] Closing connection"
end

puts "[#{pid}] Shutting down"

File: pub-sub/Subscriber.java
import com.rabbitmq.client.*;
import java.io.IOException;
import java.util.Scanner;

public class Subscriber {
    private static final String EXCHANGE_NAME = "logs";

    public static void main(String[] argv) throws Exception {

        Scanner s = new Scanner(System.in);
        System.out.print("[Java] Enter server IP (localhost)");
        String host = s.nextLine();
        if(host.isEmpty()) {
            host = "localhost";
        }

        ConnectionFactory factory = new ConnectionFactory();
        factory.setHost(host);
        Connection connection = factory.newConnection();
        Channel channel = connection.createChannel();

        channel.exchangeDeclare(EXCHANGE_NAME, "fanout");
        String queueName = channel.queueDeclare().getQueue();
        channel.queueBind(queueName, EXCHANGE_NAME, "");

        System.out.println("[Java] Waiting for messages. To exit press CTRL+C");

        Consumer consumer = new DefaultConsumer(channel) {
            @Override
            public void handleDelivery(String consumerTag, Envelope envelope,
                                       AMQP.BasicProperties properties, byte[] body) throws IOException {
                String message = new String(body, "UTF-8");
                System.out.println("[Java] Received '" + message + "'");
            }
        };
        channel.basicConsume(queueName, true, consumer);
    }
}
